pinot-core/src/test/java/org/apache/pinot/queries/DistinctCountThetaSketchTest.java

```diff
@@ -117,60 +121,77 @@ public void testGroupBySql() {
     testThetaSketches(true, true);
   }
 
+  @Test(expectedExceptions = BadQueryRequestException.class, dataProvider = "badQueries")
+  public void testInvalidNoPredicates(final String query) {
+    getBrokerResponseForSqlQuery(query);
+  }
+
+  @DataProvider(name = "badQueries")
+  public Object[][] badQueries() {
+    return new Object[][] {
+        // need at least 4 arguments in agg func
+        {"select distinctCountThetaSketch(colTS, 'nominalEntries=123', '$0') from testTable"},
+        // substitution arguments should start at $1
+        {"select distinctCountThetaSketch(colTS, 'nominalEntries=123', 'colA = 1', '$0') from testTable"},
+        // substituting variable has numeric value higher than the number of predicates provided
+        {"select distinctCountThetaSketch(colTS, 'nominalEntries=123', 'colA = 1', '$5') from testTable"},
+        // SET_DIFF requires exactly 2 arguments
+        {"select distinctCountThetaSketch(colTS, 'nominalEntries=123', 'colA = 1', 'SET_DIFF($1)') from testTable"},
+        // invalid merging function
+        {"select distinctCountThetaSketch(colTS, 'nominalEntries=123', 'colA = 1', 'asdf') from testTable"}
+    };
+  }
+
   private void testThetaSketches(boolean groupBy, boolean sql) {
     String tsQuery, distinctQuery;
     String thetaSketchParams = "nominalEntries=1001";
 
     List<String> predicateStrings = Collections.singletonList("colA = 1");
+    String substitution = "$1";
     String whereClause = Strings.join(predicateStrings, " or ");
-    tsQuery = buildQuery(whereClause, thetaSketchParams, predicateStrings, whereClause, groupBy, false);
+    tsQuery = buildQuery(whereClause, thetaSketchParams, predicateStrings, substitution, groupBy, false);
```

Comment:
Isn't the fourth argument postAggregationExpression? If so, it should look more like a set operation, as opposed to "$1"?